R/rows-db.R

```diff
@@ -423,13 +445,14 @@ sql_rows_delete.tbl_sql <- function(x, y, by, ..., returning_cols = NULL) {
   p <- sql_rows_update_prep(x, y, by)
 
   sql <- paste0(
-    "WITH ", p$y_name, "(", p$y_columns_qq, ") AS (\n",
-    dbplyr::sql_render(y),
-    "\n)\n",
-    #
     "DELETE FROM ", p$name, "\n",
     sql_output_cols(x, returning_cols, delete = TRUE),
-    "WHERE EXISTS (SELECT * FROM ", p$y_name, " WHERE ", p$compare_qual_qq, ")",
+    "WHERE EXISTS (\n",
+    "  SELECT * FROM (\n",
+    "    ", dbplyr::sql_render(y), "\n",
```

Comment:
What's the motivation for this change? How/why did it work before?

Reply:
This just replaces the CTE with a subquery to make it work with DuckDB (CTEs don't work with INSERT, UPDATE, DELETE...)